### PR TITLE
Change Revision to Version

### PR DESCRIPTION
**Pull request by pzalep1**
Please ignore all of the merge master commits, I'll do a squash and merge. Something is being weird with this branch and git. 

This PR updates the revision property to version and the hasRevision property to revisionURI.

### CODE REVIEW

File: src/FileManager/interfaces/FileManager.ts [hunk 1]
```diff
@@ -8,15 +8,15 @@ export abstract class FileManager {
    * @abstract
    * @param {string} authorUsername [The Learning Object's author's username]
    * @param {string} learningObjectId [The id of the Learning Object to upload file to]
-   * @param {number} learningObjectRevisionId [The revision id of the Learning Object]
+   * @param {number} version [The version id of the Learning Object]
```
Comment (nvisal1) [inline src/FileManager/interfaces/FileManager.ts]:
```suggestion
   * @param {number} version [The version of the Learning Object]
```

File: src/FileManager/interfaces/FileManager.ts [hunk 2]
```diff
@@ -26,15 +26,15 @@ export abstract class FileManager {
...
```
Comment (nvisal1) [inline src/FileManager/interfaces/FileManager.ts]:
```suggestion
   * @param {number} version [The version of the Learning Object]
```

File: src/FileManager/interfaces/FileManager.ts [hunk 3]
```diff
@@ -44,15 +44,15 @@ export abstract class FileManager {
...
```
Comment (nvisal1) [inline src/FileManager/interfaces/FileManager.ts]:
```suggestion
   * @param {number} version [The version of the Learning Object]
```

File: src/FileManager/interfaces/FileManager.ts [hunk 4]
```diff
@@ -62,15 +62,15 @@ export abstract class FileManager {
...
```
Comment (nvisal1) [inline src/FileManager/interfaces/FileManager.ts]:
```suggestion
   * @param {number} version [The version of the Learning Object]
```

File: src/FileManager/interfaces/FileManager.ts [hunk 5]
```diff
@@ -81,15 +81,15 @@ export abstract class FileManager {
...
```
Comment (nvisal1) [inline src/FileManager/interfaces/FileManager.ts]:
```suggestion
   * @param {number} version [The version of the Learning Object]
```

File: src/FileMetadata/interfaces/FileManagerGateway.ts [hunk 6]
```diff
@@ -5,14 +5,14 @@ export abstract class FileManagerGateway {
    * @abstract
    * @param {string} authorUsername [The Learning Object's author's username]
    * @param {string} learningObjectId [The id of the Learning Object to upload file to]
-   * @param {number} learningObjectRevisionId [The revision id of the Learning Object]
+   * @param {number} version [The version id of the Learning Object]
```
Comment (nvisal1) [inline src/FileMetadata/interfaces/FileManagerGateway.ts]:
```suggestion
   * @param {number} version [The version of the Learning Object]
```

File: src/FileMetadata/interfaces/FileManagerGateway.ts [hunk 7]
```diff
@@ -22,14 +22,14 @@ export abstract class FileManagerGateway {
...
```
Comment (nvisal1) [inline src/FileMetadata/interfaces/FileManagerGateway.ts]:
```suggestion
   * @param {number} version [The version of the Learning Object]
```

File: src/LearningObjectSearch/interactors/searchUsersObjects/searchUsersObjects.ts [hunk 8]
```diff
@@ -47,7 +48,7 @@ const GATEWAY_API = process.env.GATEWAY_API;
  * *** NOTES ***
  * If the specified user cannot be found, a NotFound ResourceError is thrown.
  * Only the author and privileged users are allowed to view Learning Object drafts.
- * "Drafts" are defined as 'not released' Learning Objects that have never been released or have a `revision` id of `0`, so
+ * "Drafts" are defined as 'not released' Learning Objects that have never been released or have a `version` id of `0`, so
```
Comment (nvisal1) [inline src/LearningObjectSearch/interactors/searchUsersObjects/searchUsersObjects.ts]:
```suggestion
 * "Drafts" are defined as 'not released' Learning Objects that have never been released or have a `version` of `0`, so
```

File: src/LearningObjects/Publishing/Bundler/Interactor.ts [hunk 9]
```diff
@@ -100,21 +100,21 @@ async function addCCLicense(prefix: string = ''): Promise<BundleData> {
  *
  *  @param {string} authorUsername [The username of the Learning Object's author]
  * @param {string} learningObjectId [The id of the Learning Object]
- * @param {number} learningObjectRevisionId [The revision id of the Learning Object]
+ * @param {number} version [The version id of the Learning Object]
```
Comment (nvisal1) [inline src/LearningObjects/Publishing/Bundler/Interactor.ts]:
```suggestion
 * @param {number} version [The version of the Learning Object]
```

File: src/LearningObjects/Publishing/Bundler/Interactor.ts [hunk 10]
```diff
@@ -136,21 +136,21 @@ async function addReadMe({
  *
  * @param {string} authorUsername [The username of the Learning Object's author]
  * @param {string} learningObjectId [The id of the Learning Object]
- * @param {number} learningObjectRevisionId [The revision id of the Learning Object]
+ * @param {number} version [The version id of the Learning Object]
```
Comment (nvisal1) [inline src/LearningObjects/Publishing/Bundler/Interactor.ts]:
```suggestion
 * @param {number} version [The version of the Learning Object]
```

File: src/LearningObjects/Publishing/Bundler/interfaces/FileGateway.ts [hunk 11]
```diff
@@ -8,15 +8,15 @@ export abstract class FileGateway implements Partial<FileManagerModule> {
    * @abstract
    * @param {string} authorUsername [The username of the Learning Object's author]
    * @param {string} learningObjectId [The id of the Learning Object]
-   * @param {number} learningObjectRevisionId [The revision id of the Learning Object]
+   * @param {number} version [The version id of the Learning Object]
```
Comment (nvisal1) [inline src/LearningObjects/Publishing/Bundler/interfaces/FileGateway.ts]:
```suggestion
   * @param {number} version [The version of the Learning Object]
```

File: src/LearningObjects/Publishing/FileManagerGateway.ts [hunk 12]
```diff
@@ -7,15 +7,15 @@ export abstract class FileManagerGateway {
    * @abstract
    * @param {string} authorUsername [The Learning Object's author's username]
    * @param {string} learningObjectId [The id of the Learning Object to upload file to]
-   * @param {number} learningObjectRevisionId [The revision id of the Learning Object]
+   * @param {number} version [The version id of the Learning Object]
```
Comment (nvisal1) [inline src/LearningObjects/Publishing/FileManagerGateway.ts]:
```suggestion
   * @param {number} version [The version of the Learning Object]
```

File: src/LearningObjects/Revisions/interactors/createLearningObjectRevision.ts [hunk 13]
```diff
@@ -60,31 +60,31 @@ export async function createLearningObjectRevision(params: {
...
+  return version;
 }
 
 /**
  * determineRevisionType uses information about the requestor to determine if
  * the revision should be placed in control of the Author or Editorial Team.
  * @param params.releasedCopy the summary information of the released Learning Object
  * @param params.learningObjectId the unique identifier of the Learning Object being revised
- * @param params.newRevisionId new revision id to be created
+ * @param params.version new version id to be created
```
Comment (nvisal1) [inline src/LearningObjects/Revisions/interactors/createLearningObjectRevision.ts]:
```suggestion
 * @param params.version new version to be created
```

File: src/LearningObjects/Revisions/interactors/createLearningObjectRevision.ts [hunk 14]
```diff
@@ -112,7 +112,7 @@ async function determineRevisionType(params: {
  * Working Copy.
  *
  * @param params.revisionStatus the status to start the revision at
- * @param params.newRevisionId new revision id to be created
+ * @param params.version new version id to be created
```
Comment (nvisal1) [inline src/LearningObjects/Revisions/interactors/createLearningObjectRevision.ts]:
```suggestion
 * @param params.version new version to be created
```

File: src/LearningObjects/interfaces/FileManagerGateway.ts [hunk 15]
```diff
@@ -7,14 +7,14 @@ export abstract class FileManagerGateway {
    * @abstract
    * @param {string} authorUsername [The Learning Object's author's username]
    * @param {string} learningObjectId [The id of the Learning Object to upload file to]
-   * @param {number} learningObjectRevisionId [The revision id of the Learning Object]
+   * @param {number} version [The version id of the Learning Object]
```
Comment (nvisal1) [inline src/LearningObjects/interfaces/FileManagerGateway.ts]:
```suggestion
   * @param {number} version [The version of the Learning Object]
```

File: src/LearningObjects/interfaces/FileManagerGateway.ts [hunk 16]
```diff
@@ -24,14 +24,14 @@ export abstract class FileManagerGateway {
...
```
Comment (nvisal1) [inline src/LearningObjects/interfaces/FileManagerGateway.ts]:
```suggestion
   * @param {number} version [The version of the Learning Object]
```

File: src/LearningObjects/interfaces/FileManagerGateway.ts [hunk 17]
```diff
@@ -41,14 +41,14 @@ export abstract class FileManagerGateway {
...
```
Comment (nvisal1) [inline src/LearningObjects/interfaces/FileManagerGateway.ts]:
```suggestion
   * @param {number} version [The version of the Learning Object]
```

File: src/shared/MongoDB/HelperFunctions/learningObjectHasRevision/learningObjectHasRevision.ts [hunk 18]
```diff
@@ -6,15 +6,15 @@ import { LearningObjectState } from '../../../types';
 /**
  * Checks if Learning Object in released collection has a copy in the objects collection that has a status of not released
  *
- * @param {string} learningObjectId [The id of the Learning Object to check for an existing revision of]
+ * @param {string} version [The id of the Learning Object to check for an existing version of]
```
Comment (nvisal1) [inline src/shared/MongoDB/HelperFunctions/learningObjectHasRevision/learningObjectHasRevision.ts]: This param seems misplaced.

Comment (pzalep1) [reply src/shared/MongoDB/HelperFunctions/learningObjectHasRevision/learningObjectHasRevision.ts]: It's supposed to be learningObjectCUID.

File: src/shared/MongoDB/HelperFunctions/index.ts [hunk 19]
```diff
@@ -2,7 +2,6 @@ import { documentLearningObject } from './documentLearningObject/documentLearnin
 import { documentReleasedLearningObject } from './documentReleasedLearningObject/documentReleasedLearningObject';
 import { generateLearningObject } from './generateLearningObject/generateLearningObject';
 import { generateReleasedLearningObject } from './generateReleasedLearningObject/generateReleasedLearningObject';
-import { generateReleasedLearningObjectSummary } from './generateReleasedLearningObjectSummary/generateReleasedLearningObjectSummary';
```
Comment (nrwinner) [inline src/shared/MongoDB/HelperFunctions/index.ts]: Why was this removed?

File: src/shared/MongoDB/HelperFunctions/learningObjectHasRevision/learningObjectHasRevision.ts [hunk 20]
```diff
@@ -31,14 +31,15 @@ export async function learningObjectHasRevision(
           connectToField: 'cuid',
           as: 'workingCopy',
           restrictSearchWithMatch: {
-            status: LearningObject.Status.RELEASED,
```
Comment (nrwinner) [inline src/shared/MongoDB/HelperFunctions/learningObjectHasRevision/learningObjectHasRevision.ts]: This is probably correct, but I'm having a little trouble wrapping my head around this change.

File: src/shared/MongoDB/HelperFunctions/learningObjectHasRevision/learningObjectHasRevision.ts [hunk 21]
```diff
@@ -31,14 +31,15 @@ export async function learningObjectHasRevision(
...
+            status: { $in: LearningObjectState.NOT_RELEASED },
           },
         },
       },
       { $unwind: { path: '$workingCopy', preserveNullAndEmptyArrays: true } },
       {
         $match: {
-          'workingCopy.status': { $in: LearningObjectState.IN_REVIEW },
```
Comment (nrwinner) [inline src/shared/MongoDB/HelperFunctions/learningObjectHasRevision/learningObjectHasRevision.ts]: Why are we including un-submitted objects here? Again, this is probably correct and I just don't understand.

File: src/shared/MongoDB/HelperFunctions/learningObjectHasRevision/learningObjectHasRevision.ts [hunk 22]
```diff
@@ -49,12 +50,15 @@ export async function learningObjectHasRevision(
       {
         $project: {
           _id: 1,
+          status: 1,
         },
       },
     ])
     .toArray();
-  if (revision) {
+  if (revision.length > 0) {
```
Comment (nrwinner) [inline src/shared/MongoDB/HelperFunctions/learningObjectHasRevision/learningObjectHasRevision.ts]: Is `revision` an array here?

File: src/shared/MongoDB/HelperFunctions/learningObjectHasRevision/learningObjectHasRevision.ts [hunk 23]
```diff
@@ -49,12 +50,15 @@ export async function learningObjectHasRevision(
...
     return true;
+  } else if (revision.length === 0) {
+    return false;
   }
-  return false;
+  throw new Error('false');
```
Comment (nrwinner) [inline src/shared/MongoDB/HelperFunctions/learningObjectHasRevision/learningObjectHasRevision.ts]: image

File: src/shared/types/learning-object-state.ts [hunk 24]
```diff
@@ -11,6 +11,13 @@ export const LearningObjectState = {
     LearningObject.Status.PROOFING,
   ],
   RELEASED: [LearningObject.Status.RELEASED],
+  NOT_RELEASED: [
```
Comment (nrwinner) [inline src/shared/types/learning-object-state.ts]: We could probably do something neat with Typescript here and union types instead of redefining. Not worried about it, just wanna be a cool kid.
welcome-to-the-cool-kids-club-et-eesen-welcome-to-52783285

File: src/shared/MongoDB/HelperFunctions/index.ts (hunk 19 above)
Comment (pzalep1) [reply src/shared/MongoDB/HelperFunctions/index.ts]: It is not being used anywhere

File: src/shared/MongoDB/HelperFunctions/learningObjectHasRevision/learningObjectHasRevision.ts (hunk 21 above)
Comment (pzalep1) [reply src/shared/MongoDB/HelperFunctions/learningObjectHasRevision/learningObjectHasRevision.ts]: When we're querying to see if a learning object has a revision of any state the ones that were in unreleased were resulting in the revision property being set to null which would not work for the builder if when were are determining if an editor can create a revision or not.

File: src/shared/MongoDB/HelperFunctions/learningObjectHasRevision/learningObjectHasRevision.ts (hunk 22 above)
Comment (pzalep1) [reply src/shared/MongoDB/HelperFunctions/learningObjectHasRevision/learningObjectHasRevision.ts]: It is a cursor that is then set to an array.

File: src/shared/MongoDB/HelperFunctions/learningObjectHasRevision/learningObjectHasRevision.ts (hunk 23 above)
Comment (pzalep1) [reply src/shared/MongoDB/HelperFunctions/learningObjectHasRevision/learningObjectHasRevision.ts]: Whoops. Should not be there.
